### ISSUE TITLE
Exciting prospects ahead!

### ISSUE THREAD

**Issue by Danimations** [LeifBloomquist/VICMIDI#1]
After reading the documentation and hearing your demo, @LeifBloomquist , I'm keen to support this project and pitch in as best I can. I don't have a VIC-20 at moment (I should be able to source one soon), and I've never programmed anything with MIDI or anything for the VIC-20, but I'm keen to have a look at the files here and chat through possibilities. I dabbled with BASIC in the 1980s and have done some programming more recently.

As an electronic musician, I would love any extra expressive CC commands, like modulation and pitchbending if you think these could be practical to implement. Is there a VIC-20 emulator I can use to playtest the existing code? And if so, can I send MIDI controllers signals to it somehow?

Count me in as keen to help take this project further. I'm also an illustrator, so if you'd like to jazz up the project artwork, manual, etc. I have skills to share. I'd gladly provide these services in return for a cart, if they're of value to you (Jim?)... and I'd like to make a proper video review of the cartridge, after being a bit underwhelmed by the only one I could find on Youtube (the conference talk videos circa 2015 were more inspiring). I'd do both of these things in return for a freebie. :P

I have a general "Danimations Digital Media Tips" channel on Youtube, where I'd like to showcase the project and you'll hear a little bit of my chiptune compositional spark in the theme music there too. ;) http://youtube.com/danimationstips

**Comment by LeifBloomquist** [comment]
Great!  Sounds good to me.  There's definitely more enhancements that could be done.  

Just keep in mind the VIC voices are *very* limited, each one has a possible 127 fixed frequencies and that's all.   There is an obscure phenomenon called "viznut's waveforms" that adds some extra crunch to the voices but it's tricky to implement and I kind of gave up on it.

As for emulators, there is indeed an emulator called VICE (https://vice-emu.sourceforge.io/) and the latest version does have MIDI support - one of the options is "Maplin" which is what the VIC-MIDI was based on.   So it might work, though I'm not sure how you would actually send MIDI data to it.

Also keep in mind the MIDI handling code is pretty raw assembly language, you might need to brush up :-)

**Comment by Danimations** [comment]
Cool, who knows whether a USB MIDI interface for Windows will talk to VICE and allow me to send signals from a MIDI keyboard? Will have to test and see. The programming aspect is probably beyond me, at least in the short-medium term. Is Paul Slocum (of Cynthcart fame) still on the scene? Is he aware of this project?

**Comment by LeifBloomquist** [comment]
Yeah, I'm not sure how VICE does it - for the emulated modems it uses a TCP/IP socket for instance.

Haven't talked to Paul in years, though Cynthcart was one of the inspirations!
